File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int>temp;
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
     sort(nums.begin(),nums.end());
     r(nums,target,0);
     return ans;   
    }
    void r(vector<int>a,int t,int s ){
        if(t==0){
            ans.push_back(temp);
            return;
        }
        for(int i=s;i<a.size();i++){
            if(t-a[i]<0)return;
            temp.push_back(a[i]);
            r(a,t-a[i],i);
            temp.pop_back();
        }
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        a={}
        ans=[]
        nums.sort()
        for x in nums:
            a[x]=a.get(x,0)+1
        for (x,y) in a.items():
            # if a[x] >= k:
                ans.append([y,x])
        ans.sort()
        res=[]
        while len(res)<k:
            res.append(ans.pop()[1])
        return res
    